File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/ext_war_unit__submitted.f90
subroutine ext_war_unit_fp64(a, b, LEN_1D, workspace, workspace_size) bind(C, name="ext_war_unit_fp64")
  use iso_c_binding
  use omp_lib
  implicit none
  ! Arguments: double* a, double* b, int64_t LEN_1D, uint8_t* workspace, int64_t workspace_size
  real(c_double), intent(inout) :: a(*)
  real(c_double), intent(in) :: b(*)
  integer(c_int64_t), value, intent(in) :: LEN_1D
  type(c_ptr), value :: workspace
  integer(c_int64_t), value :: workspace_size
  integer(c_int64_t) :: i

  if (LEN_1D > 1) then
    !$omp simd
    do i = 1, LEN_1D-1
      a(i) = a(i+1) + b(i)
    end do
    ! No explicit end directive needed for !$omp simd
  end if
end subroutine ext_war_unit_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c/ext_war_unit__submitted.c
#include <stdint.h>

void ext_war_unit_fp64(double *restrict a, double *restrict b, int64_t LEN_1D,
                       uint8_t *restrict workspace, int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;
    const int64_t n = LEN_1D - 1;
    double *aa = __builtin_assume_aligned(a, 32);
    double *bb = __builtin_assume_aligned(b, 32);
    for (int64_t i = 0; i < n; ++i) {
        aa[i] = aa[i + 1] + bb[i];
    }
}
